### PR TITLE
fix: Validate boolean query params

### PR DESCRIPTION
**Pull request by olivermrbl**
Closes CC-70

### CODE REVIEW

File: integration-tests/http/__tests__/product/admin/product.spec.ts [hunk 1]
```diff
@@ -1092,6 +1092,58 @@ medusaIntegrationTestRunner({
             }),
           ])
         })
+
+        it.only("should get product variants filtered by manage_inventory", async () => {
```
Comment (sradevski) [inline integration-tests/http/__tests__/product/admin/product.spec.ts]:
```suggestion
        it("should get product variants filtered by manage_inventory", async () => {
```

File: packages/medusa/src/api/utils/common-validators/common.ts [hunk 2]
```diff
@@ -32,7 +32,30 @@ const optionalBooleanMapper = new Map([
...
+ * Validates that a value is a boolean when it is passed as a string.
+ */
+export const booleanString = () =>
+  z
+    .union([z.boolean(), z.string()])
+    .refine((value) => {
+      if (typeof value === "boolean") {
+        return true
+      }
+      return value.toLowerCase() === "true" || value.toLowerCase() === "false"
```
Comment (sradevski) [inline packages/medusa/src/api/utils/common-validators/common.ts]:
```suggestion
      return ["true", "false"].includes(value.toString().toLowerCase())
```

File: packages/medusa/src/api/utils/common-validators/common.ts [hunk 3]
```diff
@@ -32,7 +32,30 @@ const optionalBooleanMapper = new Map([
...
+    })
+    .transform((value) => {
+      if (typeof value === "boolean") {
+        return value
+      }
+      return value.toLowerCase() === "true"
```
Comment (sradevski) [inline packages/medusa/src/api/utils/common-validators/common.ts]:
```suggestion
      return value.toString().toLowerCase() === "true"
```

**Review by sradevski — APPROVED** [review]

File: integration-tests/http/__tests__/product/admin/product.spec.ts (hunk 1 above)
Comment (olivermrbl) [reply integration-tests/http/__tests__/product/admin/product.spec.ts]: Damn

**Review by sradevski — APPROVED** [review]

**Pull request merged**